File: dificil.js
const array = [1, 1, 1, 1, 1, 2, 3, 4, 5, 6, 6, 7, 7, 7, 7, 7, 7, 7, 8, 7, 7, 8, 8, 8, 8, 9, 8, 8, 8, 8, 8]

// PRIMEIRO BLOCO DE CÓDIGO

let frequencia = {}

for (numero of array){
    if (numero in frequencia){
        //ADICIONA O NÚMERO E O SEU VALOR (SUA QUANTIDADE DE REPETIÇÕES) AO DICIONÁRIO 
        frequencia[numero] += 1 //VERIFICA SE O NÚMERO ESTA PRESENTE NO DICIONÁRIO, CASO ESTEJA E O NÚMERO APERAÇA MAIS VEZES ADICIONA MAIS 1 A SUA CONTAGEM

    } else {
        frequencia[numero] = 1 //VERIFICA SE O NÚMERO ESTA PRESENTE NO DICIONÁRIO, CASO ESTEJA E O NÚMERO NÃO SE REPETE MANTÉM SEU VALOR 
    }
}
console.log(frequencia);

// SEGUNDO BLOCO DE CÓDIGO

let quantidadeRepeticoes = 0 //INICIALIZANDO A QUANTIDADE DE REPETIÇÕES EM ZERO PARA RECEBER O VALOR POSTERIORMENTE
let numeroMaisFrequente = '' // INICIALIZANDO O NUMERO MAIS FREQUENTE COMO VAZIO/NULO PARA RECEBER O VALOR POSTERIORMENTE


//PERCORRENCO O DICIONÁRIO
for (chave in frequencia) { 
    if (frequencia[chave] > quantidadeRepeticoes ){ // VERIFICANDO SE O VALOR DA CHAVE É MAIOR DO QUE A QUANTIDADE DE REPETIÇÕES
        quantidadeRepeticoes = frequencia[chave] //CASO O VALOR SEJA MAIOR ELE SERÁ ATRIBUIDO A VARIÁVEL (quatidadeDeRepeticoes)
        numeroMaisFrequente = chave // ATRIBUI A CHAVE QUE POSSUI MAIS REPETIÇÕES NA VARIÁVEL numeroMaisFrequente
    } 
}


console.log(`Número com mais repetições: ${numeroMaisFrequente} | Quantidade de Repetições: ${quantidadeRepeticoes}`);

File: Dificil.py
# DIFICIL
# Dada um Lista de números com alguns valores repetidos, percorra a lista contando os números que mais se repetem adicionando os mesmo em um dicionário (Ex: 1: 5 - o número 2 teve 5 repetições) e exiba o número que mais se repete no dicionário
# Array: nums = [1, 1, 1, 1, 1, 2, 2, 4, 4, 5, 6, 7, 8, 6, 4, 4, 3]

# Adiciona os números em um dicionario e conta quantas vezes eles aparecem
nums = [1, 1, 1, 1, 1, 2, 2, 4, 4, 5, 6, 7, 8, 6, 4, 4, 3]

conjuntoNumeros = {}

for numero in nums:
    if numero in conjuntoNumeros:
        conjuntoNumeros[numero] += 1
    else:
        conjuntoNumeros[numero] = 1

# Pega o dicionário percorre cada elemento verificando qual tem o valor mais alto de repetições
numero_mais_frequente = ''
maior_contagem = 0

# Percorremos os itens do dicionário (chave, valor)
for numero, contagem in conjuntoNumeros.items():
    if contagem > maior_contagem:
        maior_contagem = contagem
        numero_mais_frequente = numero

print(f"O número que mais se repete é: {numero_mais_frequente}")
print(f"Ele se repete {maior_contagem} vezes.")